src/Centreon/Infrastructure/PlatformTopology/PlatformTopologyRepositoryRDB.php

```
@@ -209,4 +211,126 @@ public function findPlatform(int $serverId): ?Platform
+                $childrenPlatforms[] = EntityCreator::createEntityByArray(Platform::class, $platform);
+            }
+        }
+
+        return $childrenPlatforms;
+    }
+
+    public function updatePlatformParameters(int $serverId, array $parameters): void
+    {
+        $requestParameters = [];
+        $query = 'UPDATE `:db`.`platform_topology` SET ';
+
+        $arrayKeyLast = function ($array) {
+            return array_keys($array)[count($array) - 1];
+        };
+
+        foreach ($parameters as $column => $value) {
+            /*
+             *  Set PDO Param type.
+             */
+            switch ($column) {
+                case 'address':
+                case 'hostname':
+                case 'name':
+                case 'type':
+                    $requestParameters[':' . $column] = [
+                        \PDO::PARAM_STR => $value
+                    ];
+                    break;
+                case 'parent_id':
+                case 'server_id':
+                    $requestParameters[':' . $column] = [
+                        \PDO::PARAM_INT => $value
+                    ];
+                    break;
+            }
+
+            /**
+             * Build query.
+             */
+            if ($column === $arrayKeyLast($parameters)) {
+                $query .= "$column = :$column";
+            } else {
+                $query .= "$column = :$column, ";
+            }
+        }
+        $query .=  ' WHERE id = ' . $serverId;
+
+        try {
+            $statement = $this->db->prepare($this->translateDbName($query));
+            foreach ($requestParameters as $token => $bindValues) {
+                foreach ($bindValues as $paramType => $value) {
+                    $statement->bindValue($token, $value, $paramType);
+                }
+            }
+            $statement->execute();
+        } catch (\Exception $ex) {
+            throw new RepositoryException(_('An error occured while updating platform'));
```

@callapa should we throw an exception here